src/index.ts

```diff
@@ -252,99 +269,181 @@ function main() {
-    switchMap(([customer]) =>
-      createQuote(
-        customer,
-        cybrid.PostQuoteBankModelSideEnum.Buy,
-        'BTC-USD',
-        quantity
-      )
-    ),
-    switchMap(quote => createTrade(quote)),
-    switchMap(trade => pollTrade(trade.guid!))
+  // Create Transfer pipeline, requires Create Customer, Create Fiat Account and Create Identity
+  const executeTransferObs = createObs.pipe(
+      switchMap(([customer]) =>
+          createQuote(
+              customer,
+              cybrid.PostQuoteBankModelProductTypeEnum.BookTransfer,
+              cybrid.PostQuoteBankModelSideEnum.Deposit,
+              fiatQuantity,
+              null,
+              "USD"
+          )
+      ),
+      switchMap(quote => createTransfer(quote, cybrid.PostTransferBankModelTransferTypeEnum.Book)),
+      switchMap(transfer => pollTransfer(transfer.guid!))
   );
 
   // Get Account pipeline
-  const getAccountObs = createAccountObs.pipe(
-    combineLatestWith(executeTradeObs),
-    switchMap(([account]) => getAccount(account.guid!))
+  const getFiatAccountObs = createFiatAccountObs.pipe(
+      combineLatestWith(executeTransferObs),
+      switchMap(([account]) => getAccount(account.guid!))
   );
 
-  // Verify account balance
-  getAccountObs
+  // Verify fiat account balance
+  getFiatAccountObs
     .pipe(
-      tap(account => {
-        const balance = account.platform_balance!;
-        const expectedBalance = quantity;
-        if (balance !== expectedBalance) {
-          throw new InvalidBalanceError(balance, expectedBalance);
+        tap(account => {
+          const balance = account.platform_balance!;
+          const expectedBalance = fiatQuantity;
+          if (balance !== expectedBalance) {
+            throw new InvalidBalanceError(balance, expectedBalance);
+          }
+        })
+    );
+
+  getFiatAccountObs.subscribe({
+    next: account => {
+      const fiatBalance = account.platform_balance!;
+      console.log(`Fiat USD account for ${customer.guid} has the expected balance: ${fiatBalance}.`);
+      console.log('Test has completed successfully!');
+
+      of([customer]).pipe(
+          switchMap(([customer]) =>
+              createQuote(
+                  customer,
+                  cybrid.PostQuoteBankModelProductTypeEnum.Trading,
+                  cybrid.PostQuoteBankModelSideEnum.Buy,
+                  fiatQuantity,
+                  "BTC-USD",
+                  null
+              )
+          ),
+          switchMap(quote => createTrade(quote)),
+          switchMap(trade => pollTrade(trade.guid!)),
+          switchMap(_trade => getAccount(cryptoAccount.guid!)),
+          tap(account => {
+            const balance = account.platform_balance!;
+            const expectedBalance = cryptoQuantity;
+            if (balance !== expectedBalance) {
+              throw new InvalidBalanceError(balance, expectedBalance);
+            }
+          })
+      ).subscribe({
```

Comment:
It's not really a big deal, but typically it's not the best practice to nest subscribes. In this case it should be fine.

Reply:
Updated. ty